HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -3555,6 +3570,135 @@ def self.apply_installation_quality(model, heating_system, cooling_system, unita
+        supp_sys_power_level = [supp_sys_capacity, q_dot_defrost].min / supp_sys_efficiency # Assume perfect tempering
+      end
+    else
+      is_ducted = !heat_pump.distribution_system_idref.nil?
+      if is_ducted
+        supp_sys_fuel = heat_pump.backup_heating_fuel
+        supp_sys_capacity = UnitConversions.convert(heat_pump.backup_heating_capacity, 'Btu/hr', 'W')
+        supp_sys_efficiency = heat_pump.backup_heating_efficiency_percent
+        supp_sys_efficiency = heat_pump.backup_heating_efficiency_afue if supp_sys_efficiency.nil?
+        supp_sys_power_level = [supp_sys_capacity, q_dot_defrost].min / supp_sys_efficiency # Assume perfect tempering
+      else
+        # Practically no integrated supplemental system for ductless
+        # Sometimes integrated backup systems are added to ductless to avoid unmet loads, so it shouldn't count here to avoid overestimating backup system energy use
+        supp_sys_capacity = 0.0
+        supp_sys_power_level = 0.0
+      end
+    end
+    # other equipment actuator
+    defrost_heat_load_oed = OpenStudio::Model::OtherEquipmentDefinition.new(model)
+    defrost_heat_load_oed.setName("#{air_loop_unitary.name} defrost heat load def")
+    defrost_heat_load_oed.setDesignLevel(0)
+    defrost_heat_load_oed.setFractionRadiant(0)
+    defrost_heat_load_oed.setFractionLatent(0)
+    defrost_heat_load_oed.setFractionLost(0)
+    defrost_heat_load_oe = OpenStudio::Model::OtherEquipment.new(defrost_heat_load_oed)
+    defrost_heat_load_oe.setName("#{air_loop_unitary.name} defrost heat load")
+    defrost_heat_load_oe.setSpace(conditioned_space)
+    defrost_heat_load_oe.setSchedule(model.alwaysOnDiscreteSchedule)
+
+    defrost_heat_load_oe_act = OpenStudio::Model::EnergyManagementSystemActuator.new(defrost_heat_load_oe, *EPlus::EMSActuatorOtherEquipmentPower, defrost_heat_load_oe.space.get)
+    defrost_heat_load_oe_act.setName("#{defrost_heat_load_oe.name} act")
+
+    # TODO: handle energy output to be categorized to heating
+    energyplus_fuel = EPlus.fuel_type(supp_sys_fuel)
+    defrost_supp_heat_energy_oed = OpenStudio::Model::OtherEquipmentDefinition.new(model)
+    defrost_supp_heat_energy_oed.setName("#{air_loop_unitary.name} supp heat energy def")
+    defrost_supp_heat_energy_oed.setDesignLevel(0)
+    defrost_supp_heat_energy_oed.setFractionRadiant(0)
+    defrost_supp_heat_energy_oed.setFractionLatent(0)
+    defrost_supp_heat_energy_oed.setFractionLost(1)
+    defrost_supp_heat_energy_oe = OpenStudio::Model::OtherEquipment.new(defrost_supp_heat_energy_oed)
+    defrost_supp_heat_energy_oe.setName("#{air_loop_unitary.name} defrost supp heat energy")
+    defrost_supp_heat_energy_oe.setSpace(conditioned_space)
+    defrost_supp_heat_energy_oe.setFuelType(energyplus_fuel)
+    defrost_supp_heat_energy_oe.setSchedule(model.alwaysOnDiscreteSchedule)
+    defrost_supp_heat_energy_oe.setEndUseSubcategory(Constants.ObjectNameBackupSuppHeat)
```

Comment:
@yzhou601 @shorowit, Any thoughts on where this energy should be accounted for? Do we want to attribute it supplemental heating? E+ has a separate timeseries output for defrost but I'm not sure if that can be accessed here.

Reply:
After talking with Yueyue and Joe, this approach looks right.